README.md
# [Light Bootstrap Dashboard Angular](https://demos.creative-tim.com/light-bootstrap-dashboard-angular2/dashboard)
[![version][version-badge]][CHANGELOG] ![license][license-badge]

![alt text](src/assets/img/opt_lbd_angular_thumbnail.jpg)

**[Light Bootstrap Dashboard Angular](https://demos.creative-tim.com/light-bootstrap-dashboard-angular2/dashboard)** is an admin dashboard template designed to be beautiful and simple. It is built on top of Bootstrap 3, using [Light Bootstrap Dashboard](https://www.creative-tim.com/product/light-bootstrap-dashboard) and it is fully responsive. It comes with a big collections of elements that will offer you multiple possibilities to create the app that best fits your needs. It can be used to create admin panels, project management systems, web applications backend, CMS or CRM.

The product represents a big suite of front-end developer tools that can help you jump start your project. We have created it thinking about things you actually need in a dashboard. Light Bootstrap Dashboard Angular 2 contains multiple handpicked and optimized plugins. Everything is designed to fit with one another. As you will be able to see, the dashboard you can access on Creative Tim is a customization of this product.

It comes with 6 filter colors for the sidebar (“black”, “azure”,”green”,”orange”,”red”,”purple”) and an option to have a background image.

Special thanks go to: Robert McIntosh for the notification system Chartist for the wonderful charts We are very excited to share this dashboard with you and we look forward to hearing your feedback!

## Links:

+ [Live Preview](https://demos.creative-tim.com/light-bootstrap-dashboard-angular2/dashboard)
+ [Light Bootstrap Dashboard PRO Angular](https://www.creative-tim.com/product/light-bootstrap-dashboard-pro-angular2/?ref=lbd-angular-github) ($49)

## Quick Start:

Quick start options:

+ [Download from Github](https://github.com/creativetimofficial/light-bootstrap-dashboard-angular2/archive/master.zip).
+ [Download from Creative Tim](https://www.creative-tim.com/product/light-bootstrap-dashboard-angular2).
+ Clone the repo: `git clone https://github.com/creativetimofficial/light-bootstrap-dashboard-angular2.git`.

## Terminal Commands

1. Install NodeJs from [NodeJs Official Page](https://nodejs.org/en).
2. Open Terminal
3. Go to your file project
4. Run in terminal: ```npm install -g @angular/cli```
5. Then: ```npm install```
6. And: ```ng serve```
7. Navigate to `http://localhost:4200/`

### What's included

Within the download you'll find the following directories and files:
```
light-bootstrap-dashboard-angular
├── CHANGELOG.md
├── LICENSE.md
├── README.md
├── angular.json
├── documentation
│   ├── css
│   └── tutorial-lbd-angular2.html
├── e2e
├── karma.conf.js
├── package-lock.json
├── package.json
├── protractor.conf.js
├── src
│   ├── app
│   │   ├── app.component.css
│   │   ├── app.component.html
│   │   ├── app.component.spec.ts
│   │   ├── app.component.ts
│   │   ├── app.module.ts
│   │   ├── app.routing.ts
│   │   ├── home
│   │   │   ├── home.component.css
│   │   │   ├── home.component.html
│   │   │   ├── home.component.spec.ts
│   │   │   └── home.component.ts
│   │   ├── icons
│   │   │   ├── icons.component.css
│   │   │   ├── icons.component.html
│   │   │   ├── icons.component.spec.ts
│   │   │   └── icons.component.ts
│   │   ├── layouts
│   │   │   └── admin-layout
│   │   │       ├── admin-layout.component.html
│   │   │       ├── admin-layout.component.scss
│   │   │       ├── admin-layout.component.spec.ts
│   │   │       ├── admin-layout.component.ts
│   │   │       ├── admin-layout.module.ts
│   │   │       └── admin-layout.routing.ts
│   │   ├── lbd
│   │   │   ├── lbd-chart
│   │   │   │   ├── lbd-chart.component.html
│   │   │   │   └── lbd-chart.component.ts
│   │   │   └── lbd.module.ts
│   │   ├── maps
│   │   │   ├── maps.component.css
│   │   │   ├── maps.component.html
│   │   │   ├── maps.component.spec.ts
│   │   │   └── maps.component.ts
│   │   ├── notifications
│   │   │   ├── notifications.component.css
│   │   │   ├── notifications.component.html
│   │   │   ├── notifications.component.spec.ts
│   │   │   └── notifications.component.ts
│   │   ├── shared
│   │   │   ├── footer
│   │   │   │   ├── footer.component.html
│   │   │   │   ├── footer.component.ts
│   │   │   │   └── footer.module.ts
│   │   │   └── navbar
│   │   │       ├── navbar.component.html
│   │   │       ├── navbar.component.ts
│   │   │       └── navbar.module.ts
│   │   ├── sidebar
│   │   │   ├── sidebar.component.html
│   │   │   ├── sidebar.component.ts
│   │   │   └── sidebar.module.ts
│   │   ├── tables
│   │   │   ├── tables.component.css
│   │   │   ├── tables.component.html
│   │   │   ├── tables.component.spec.ts
│   │   │   └── tables.component.ts
│   │   ├── typography
│   │   │   ├── typography.component.css
│   │   │   ├── typography.component.html
│   │   │   ├── typography.component.spec.ts
│   │   │   └── typography.component.ts
│   │   ├── upgrade
│   │   │   ├── upgrade.component.css
│   │   │   ├── upgrade.component.html
│   │   │   ├── upgrade.component.spec.ts
│   │   │   └── upgrade.component.ts
│   │   └── user
│   │       ├── user.component.css
│   │       ├── user.component.html
│   │       ├── user.component.spec.ts
│   │       └── user.component.ts
│   ├── assets
│   │   ├── css
│   │   ├── fonts
│   │   ├── img
│   │   └── sass
│   │       ├── lbd
│   │       └── light-bootstrap-dashboard.scss
│   ├── environments
│   ├── favicon.ico
│   ├── index.html
│   ├── main.ts
│   ├── polyfills.ts
│   ├── styles.css
│   ├── test.ts
│   └── tsconfig.json
├── tslint.json
└── typings.json

```
## Useful Links

More products from Creative Tim: <https://www.creative-tim.com/bootstrap-themes>

Tutorials: <https://www.youtube.com/channel/UCVyTG4sCw-rOvB9oHkzZD1w>

Freebies: <https://www.creative-tim.com/products>

Affiliate Program (earn money): <https://www.creative-tim.com/affiliates/new>

Social Media:

Twitter: <https://twitter.com/CreativeTim>

Facebook: <https://www.facebook.com/CreativeTim>

Dribbble: <https://dribbble.com/creativetim>

Google+: <https://plus.google.com/+CreativetimPage>

Instagram: <https://instagram.com/creativetimofficial>

[CHANGELOG]: ./CHANGELOG.md

[version-badge]: https://img.shields.io/badge/version-1.5.0-blue.svg
[license-badge]: https://img.shields.io/badge/license-MIT-blue.svg


CHANGELOG.md
## [1.5.0] - 2018-10-04
### Changes
- update to Angular 6
- added online documentation 

## [1.4.0] - 2017-08-28
### Changes for Angular 4
- added angular-cli
- update to Angular 4

## [v1.3.0] 2017-08-28
### skipped for sync with Angular 4 version convention

## [v1.2.0] 2017-08-28
### skipped for sync with Angular 4 version convention

## [1.1.1] - 2017-03-21
### Added
- added "@types/core-js": "0.9.35" in package

## [1.1.0] - 2017-03-01
### Added
- fix version numbers in package.json and CSS and JS

## [1.0.2] - 2017-02-27
### Added
- added upgrade to pro page
- fixed navbar title
- changed http://www.creative-tim.com/ with https://www.creative-tim.com/

## [1.0.1] - 2017-02-13
### small fix
- changes in routes(redirect '' to 'dashboard' + added HashLocationStrategy)
- changes in documentation
- changes in sidebar(cleaned li tags + changed z-index)
- changes in footer(added current date)

## [1.0.0] - 2016-12-14
### initial Release


src/app/ag-grid/ag-grid.component.ts
import {Component, OnInit, ViewChild} from '@angular/core';
import {AllCommunityModules} from '@ag-grid-community/all-modules';
import {HttpClient} from '@angular/common/http';
import {AgGridAngular} from '@ag-grid-community/angular';
import {AllModules} from '@ag-grid-enterprise/all-modules';
import {Router} from '@angular/router';


@Component({
    selector: 'app-ag-grid',
    templateUrl: './ag-grid.component.html',
    styleUrls: ['./ag-grid.component.scss']
})
export class AgGridComponent implements OnInit {

    @ViewChild('agGrid') agGrid: AgGridAngular;

    columnDefs;
    gridApi;
    gridColumnApi;
    defaultColDef;
    rowSelection;
    paginationPageSize;
    fileName;
    sideBar;
    rowData: any = [];
    display = 'none';
    modules = AllModules;
    selectedData: any = {};
    gridOptions: any = {};

    constructor(private http: HttpClient,
                private router: Router) {


        this.columnDefs = [
            {
                field: 'athlete',
                width: 150,
                filter: 'agTextColumnFilter',
                checkboxSelection: function (params) {
                    return params.columnApi.getRowGroupColumns().length === 0;
                },
                headerCheckboxSelection: function (params) {
                    return params.columnApi.getRowGroupColumns().length === 0;
                }
            },
            {
                field: 'age',
                width: 90,
                filter: 'agNumberColumnFilter'
            },
            {
                field: 'country',
                width: 120,
                filter: 'agTextColumnFilter',
            },
            {
                field: 'year',
                width: 90,
                filter: 'agNumberColumnFilter'
            },
            {
                field: 'date',
                width: 110,
                filter: 'agTextColumnFilter',
            },
            {
                field: 'gold',
                width: 100,
                filter: 'agTextColumnFilter',
            },
            {
                field: 'silver',
                width: 100,
                filter: 'agTextColumnFilter',
            },
            {
                field: 'bronze',
                width: 100,
                filter: 'agTextColumnFilter',
            },
            {
                field: 'total',
                width: 100,
                filter: 'agNumberColumnFilter'
            }
        ];
        this.defaultColDef = {
            sortable: true,
            resizable: true,
            enableValue: true,
            enableRowGroup: true,
            enablePivot: true
        };
        this.sideBar = {
            toolPanels: [
                {
                    id: 'columns',
                    labelDefault: 'Columns',
                    labelKey: 'columns',
                    iconKey: 'columns',
                    toolPanel: 'agColumnsToolPanel',
                    toolPanelParams: {
                        suppressSyncLayoutWithGrid: true
                    }
                }
            ]
        };
        this.paginationPageSize = 15;
        this.gridOptions = {
            rowSelection: 'multiple',
            context: {
                componentParent: this
            }
        };
    }

    ngOnInit() {
    }

    getSelectedRows() {
        const selectedNodes = this.agGrid.api.getSelectedNodes();
        const selectedData = selectedNodes.map(node => node.data);
        const selectedDataStringPresentation = selectedData.map(node => node.make + ' ' + node.model).join(', ');
        alert(`Selected nodes: ${selectedDataStringPresentation}`);
    }

    onBtExport() {
        var params = {
            fileName: this.fileName
        };
        this.fileName = '';
        this.gridApi.exportDataAsCsv(params);
    }

    onGridReady(params) {
        this.gridApi = params.api;
        this.gridColumnApi = params.columnApi;
        // this.gridApi.setDomLayout('autoHeight');
        // document.querySelector('#myGrid')..height = '';
        // this.http
        //     .get("https://raw.githubusercontent.com/ag-grid/ag-grid/master/packages/ag-grid-docs/src/olympicWinners.json")
        //     .subscribe(data => {
        this.rowData = [{
            'athlete': 'Michael Phelps',
            'age': 23,
            'country': 'United States',
            'year': 2008,
            'date': '24/08/2008',
            'sport': 'Swimming',
            'gold': 8,
            'silver': 0,
            'bronze': 0,
            'total': 8
        }, {
            'athlete': 'Michael Phelps',
            'age': 19,
            'country': 'United States',
            'year': 2004,
            'date': '29/08/2004',
            'sport': 'Swimming',
            'gold': 6,
            'silver': 0,
            'bronze': 2,
            'total': 8
        }, {
            'athlete': 'Michael Phelps',
            'age': 27,
            'country': 'United States',
            'year': 2012,
            'date': '12/08/2012',
            'sport': 'Swimming',
            'gold': 4,
            'silver': 2,
            'bronze': 0,
            'total': 6
        }, {
            'athlete': 'Natalie Coughlin',
            'age': 25,
            'country': 'United States',
            'year': 2008,
            'date': '24/08/2008',
            'sport': 'Swimming',
            'gold': 1,
            'silver': 2,
            'bronze': 3,
            'total': 6
        }, {
            'athlete': 'Aleksey Nemov',
            'age': 24,
            'country': 'Russia',
            'year': 2000,
            'date': '01/10/2000',
            'sport': 'Gymnastics',
            'gold': 2,
            'silver': 1,
            'bronze': 3,
            'total': 6
        }, {
            'athlete': 'Alicia Coutts',
            'age': 24,
            'country': 'Australia',
            'year': 2012,
            'date': '12/08/2012',
            'sport': 'Swimming',
            'gold': 1,
            'silver': 3,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Missy Franklin',
            'age': 17,
            'country': 'United States',
            'year': 2012,
            'date': '12/08/2012',
            'sport': 'Swimming',
            'gold': 4,
            'silver': 0,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Ryan Lochte',
            'age': 27,
            'country': 'United States',
            'year': 2012,
            'date': '12/08/2012',
            'sport': 'Swimming',
            'gold': 2,
            'silver': 2,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Allison Schmitt',
            'age': 22,
            'country': 'United States',
            'year': 2012,
            'date': '12/08/2012',
            'sport': 'Swimming',
            'gold': 3,
            'silver': 1,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Natalie Coughlin',
            'age': 21,
            'country': 'United States',
            'year': 2004,
            'date': '29/08/2004',
            'sport': 'Swimming',
            'gold': 2,
            'silver': 2,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Ian Thorpe',
            'age': 17,
            'country': 'Australia',
            'year': 2000,
            'date': '01/10/2000',
            'sport': 'Swimming',
            'gold': 3,
            'silver': 2,
            'bronze': 0,
            'total': 5
        }, {
            'athlete': 'Dara Torres',
            'age': 33,
            'country': 'United States',
            'year': 2000,
            'date': '01/10/2000',
            'sport': 'Swimming',
            'gold': 2,
            'silver': 0,
            'bronze': 3,
            'total': 5
        }, {
            'athlete': 'Cindy Klassen',
            'age': 26,
            'country': 'Canada',
            'year': 2006,
            'date': '26/02/2006',
            'sport': 'Speed Skating',
            'gold': 1,
            'silver': 2,
            'bronze': 2,
            'total': 5
        }, {
            'athlete': 'Nastia Liukin',
            'age': 18,
            'country': 'United States',
            'year': 2008,
            'date': '24/08/2008',
            'sport': 'Gymnastics',
            'gold': 1,
            'silver': 3,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Marit Bjørgen',
            'age': 29,
            'country': 'Norway',
            'year': 2010,
            'date': '28/02/2010',
            'sport': 'Cross Country Skiing',
            'gold': 3,
            'silver': 1,
            'bronze': 1,
            'total': 5
        }, {
            'athlete': 'Sun Yang',
            'age': 20,
            'country': 'China',
            'year': 2012,
            'date': '12/08/2012',
            'sport': 'Swimming',
            'gold': 2,
            'silver': 1,
            'bronze': 1,
            'total': 4
        }, {
            'athlete': 'Kirsty Coventry',
            'age': 24,
            'country': 'Zimbabwe',
            'year': 2008,
            'date': '24/08/2008',
            'sport': 'Swimming',
            'gold': 1,
            'silver': 3,
            'bronze': 0,
            'total': 4
        }, {
            'athlete': 'Libby Lenton-Trickett',
            'age': 23,
            'country': 'Australia',
            'year': 2008,
            'date': '24/08/2008',
            'sport': 'Swimming',
            'gold': 2,
            'silver': 1,
            'bronze': 1,
            'total': 4
        }, {
            'athlete': 'Ryan Lochte',
            'age': 24,
            'country': 'United States',
            'year': 2008,
            'date': '24/08/2008',
            'sport': 'Swimming',
            'gold': 2,
            'silver': 0,
            'bronze': 2,
            'total': 4
        }];
        // });
    }

    rowsSelected() {
        return this.gridApi && this.gridApi.getSelectedRows().length > 0;
    }

    selectedRows(val) {
        const selectRows = this.gridApi.getSelectedRows();
        console.log('selected row', selectRows, selectRows.length)
        if (selectRows.length === 0) {
            alert('Select any row');
        } else {
            if(val === 'close_alert') {
                document.getElementById('close_popup').style.display = 'block';
            } else if (val === 'route_alert') {
                document.getElementById('route_popup').style.display = 'block';
            } else if (val === 'add_case') {
                document.getElementById('add_case').style.display = 'block';
            } else if (val === 'check_selected') {
                alert('Selected Alerts Checked In')
            }
        }
    }

    editRow() {
        this.display = 'block';
        this.selectedData = this.gridApi.getSelectedRows()[0];
        console.log('selected row', this.selectedData);
    }

    onCloseHandled() {
        this.display = 'none';
        document.getElementById('close_popup').style.display  = 'none';
        document.getElementById('route_popup').style.display  = 'none';
        document.getElementById('add_case').style.display  = 'none';
    }

    onCellDoubleClicked(e) {
        console.log('event', e);
        this.router.navigate(['/dashboard']);
    }

    getContextMenuItems(params) {
        var result = [
            {
                name: 'Close Selected',
                action: function () {
                    document.getElementById('close_popup').style.display = 'block'
                },
            },
            {
                name: 'Route Selected',
                action: function() {
                    document.getElementById('route_popup').style.display = 'block'
                },
            },
            {
                name: 'Add to case',
                action: function() {
                    document.getElementById('add_case').style.display = 'block'
                },
            },
            'copy',
        ];
        return result;
    }

}
